addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TimestampDisclosureScanRule.java

```diff
@@ -138,28 +140,13 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         }
         LOGGER.debug("Checking message {} for timestamps", msg.getRequestHeader().getURI());
 
-        List<HttpHeaderField> responseheaders = msg.getResponseHeader().getHeaders();
-        StringBuilder filteredResponseheaders = new StringBuilder();
-        for (HttpHeaderField responseheader : responseheaders) {
-            boolean ignoreHeader = false;
-            for (String headerToIgnore : RESPONSE_HEADERS_TO_IGNORE) {
-                if (responseheader.getName().equalsIgnoreCase(headerToIgnore)) {
-                    LOGGER.debug("Ignoring header {}", responseheader.getName());
-                    ignoreHeader = true;
-                    break; // out of inner loop
-                }
-            }
-            if (!ignoreHeader) {
-                filteredResponseheaders
-                        .append('\n')
-                        .append(responseheader.getName())
-                        .append(": ")
-                        .append(responseheader.getValue());
-            }
-        }
-
-        String responsebody = msg.getResponseBody().toString();
-        String[] responseparts = {filteredResponseheaders.toString(), responsebody};
+        List<HttpHeaderField> responseparts = new ArrayList<>();
+        responseparts.addAll(
+                msg.getResponseHeader().getHeaders().stream()
+                        .filter(header -> !RESPONSE_HEADERS_TO_IGNORE.contains(header.getName()))
```

Comment:
Needs a test for the headers in `RESPONSE_HEADERS_TO_IGNORE` but with different case.